### PR TITLE
feat: add takeCanisterSnapshot to ic-mgmt

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Add support for [take_canister_snapshot](https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-take_canister_snapshot) to ic-mgmt.

# Changes

- Implement `takeCanisterSnapshot`
- Create a type `SnapshotId` for the string representation
- Expose related did types in index for the consumer

### CODE REVIEW

File: packages/ic-management/src/ic-management.canister.ts [hunk 1]
```diff
@@ -350,4 +355,41 @@ export class ICManagementCanister {
...
+  }: {
+    canisterId: Principal;
+    snapshotId?: SnapshotIdText | snapshot_id;
+  }): Promise<take_canister_snapshot_result> => {
+    const { take_canister_snapshot } = this.service;
+
+    return take_canister_snapshot({
+      canister_id: canisterId,
+      replace_snapshot: toNullable(
+        nonNullish(snapshotId)
```
Comment (AntonioVentilii) [inline packages/ic-management/src/ic-management.canister.ts:387]: FUTURE development: maybe this part of parsing `SnapshotIdText | snapshot_id` can go in another small util (for another PR obviously), like something to parse `SnapshotIdText | snapshot_id` to the correct type always, for example `snapshot_id` only

File: packages/ic-management/src/types/ic-management.params.ts [hunk 2]
```diff
@@ -132,3 +132,5 @@ export interface ProvisionalTopUpCanisterParams {
   canisterId: Principal;
   amount: bigint;
 }
+
+export type SnapshotIdText = string;
```
Comment (AntonioVentilii) [inline packages/ic-management/src/types/ic-management.params.ts:136]: this type is not exported in `index`?

**Review by AntonioVentilii — APPROVED** [review]
LGTM, tks! 2 small observations but not blockers, I think

File: packages/ic-management/src/ic-management.canister.ts (hunk 1 above)
Comment (peterpeterparker) [reply packages/ic-management/src/ic-management.canister.ts:387]: Good idea. There is 2-3 other functions to add too, let's see if there is also some things that should be reused between those.

File: packages/ic-management/src/types/ic-management.params.ts (hunk 2 above)
Comment (peterpeterparker) [reply packages/ic-management/src/types/ic-management.params.ts:136]: In `index`, this all module, `ic-management.params` is exposed.

```
...
export * from "./types/ic-management.params";
...
```

Comment (AntonioVentilii) [reply packages/ic-management/src/types/ic-management.params.ts:136]: perfect tks!

**Pull request merged**